src/sinks/splunk_hec/logs/encoder.rs

```diff
@@ -72,29 +75,49 @@ impl Encoder<Vec<HecProcessedEvent>> for HecLogsEncoder {
 
                 let mut bytes = BytesMut::new();
                 let serializer = encoder.serializer();
-                let hec_event = if serializer.supports_json() {
-                    HecEvent::Json(
-                        serializer
-                            .to_json_value(event)
-                            .map_err(|error| emit!(SplunkEventEncodeError { error }))
-                            .ok()?,
-                    )
-                } else {
-                    encoder.encode(event, &mut bytes).ok()?;
-                    HecEvent::Text(String::from_utf8_lossy(&bytes))
-                };
 
-                let mut hec_data = HecData::new(hec_event, metadata.fields, metadata.timestamp);
-                hec_data.host = metadata.host.map(|host| host.to_string_lossy());
-                hec_data.index = metadata.index;
-                hec_data.source = metadata.source;
-                hec_data.sourcetype = metadata.sourcetype;
+                match metadata.endpoint_target {
+                    EndpointTarget::Raw => {
+                        if serializer.supports_json() {
+                            match serde_json::to_vec(&event) {
+                                Ok(value) => Some(value),
+                                Err(error) => {
+                                    emit!(SplunkEventEncodeError { error });
+                                    None
+                                }
+                            }
+                        } else {
+                            encoder.encode(event, &mut bytes).ok()?;
+                            Some(bytes.to_vec())
+                        }
+                    }
```

Comment:
Do we need the `supports_json()` check here? It seems like we encode to a Vec either way. I think it was done before to create separate `HecEvent` variants depending on the input.

Reply:
Yes good catch. Just `encoder.encode` handles everything we need.